File: cpp/programmers/level_0/짝수의 합.cpp
//
//  짝수의 합.cpp
//  cpp
//
//  Created on 2023/01/22.
//

#include <stdio.h>

#include <string>
#include <vector>


using namespace std;

int solution(int n) {
    int sum = 0;
    int i = 0;
    
    while (i <= n) {
        sum += i;
        
        i += 2;
    }
    
    
    return sum;
}

// debug
int main(void) {
    printf("Test Case: %d\n", solution(10)); // 30
    printf("Test Case: %d\n", solution(4)); // 6
    return 0;
}

File: java/programmers/level_0/src/짝수의_합.java
public class 짝수의_합 {
//정수 n이 주어질 때, n이하의 짝수를 모두 더한 값을 return 하도록 solution 함수를 작성해주세요.
  public static int solution(int n) {
    int answer = 0;

    for (int i = 1; i <= n; i++) {
      if (i % 2 == 0) {   //n이 짝수라면
        answer += i;
      }
    }
    return answer;
  }

  public static void main(String[] args) {
    System.out.println(solution(16));
  }
}
